### PR TITLE
Use tibble package

### LINKED ISSUE #1488: Extract functions related to the tbl_df class to a separate package
Other packages that are now creating data frames of class `tbl_df` could import that package to make use of it, or use `data_frame()` or `tibble()` without having to depend on `dplyr`. I have a [proof of concept](https://github.com/krlmlr/tibble), and would be happy to implement the split -- perhaps after the next `dplyr` release.

This is how the `NAMESPACE` could look like (except that `as_data_frame` is now a generic).

```
S3method("[",tbl_df)
S3method(as.data.frame,tbl_df)
S3method(obj_type,"NULL")
S3method(obj_type,data.frame)
S3method(obj_type,data_frame)
S3method(obj_type,default)
S3method(print,BoolResult)
S3method(print,tbl_df)
S3method(print,trunc_mat)
S3method(type_sum,Date)
S3method(type_sum,POSIXt)
S3method(type_sum,array)
S3method(type_sum,character)
S3method(type_sum,data.frame)
S3method(type_sum,default)
S3method(type_sum,factor)
S3method(type_sum,integer)
S3method(type_sum,logical)
S3method(type_sum,matrix)
S3method(type_sum,numeric)
export(as.tbl.data.frame)
export(as_data_frame)
export(data_frame)
export(data_frame_)
export(dim_desc)
export(frame_data)
export(knit_print.trunc_mat)
export(tbl_df)
export(tibble)
export(trunc_mat)
export(type_sum)
```

### PR DESCRIPTION
**Pull request by krlmlr**
Reexports all functions exported by tibble.

Fixes #1488.

A few tests (of functions now private to tibble) had to be removed. One test had to be adapted, because `names(data_frame())` is now `character(0L)` instead of `NULL`.

### CODE REVIEW

**Comment by krlmlr** [comment]
Updated, now leaves src() and tbl() related functions in this package; compatible with tibble 0.1-5.

File: tests/testthat/test-data_frame.R [hunk 1]
```diff
@@ -38,21 +38,21 @@ test_that("add_rownames keeps the tbl classes (#882)", {
 
 test_that("2d object isn't a valid column", {
   expect_error(
-    check_data_frame(list(x = mtcars)),
+    tibble:::check_data_frame(list(x = mtcars)),
```
Comment (hadley) [inline tests/testthat/test-data_frame.R]: Maybe these tests should just be deleted?

Comment (krlmlr) [inline tests/testthat/test-data_frame.R]: Yes -- keeping them for now to avoid merge conflicts with further incoming changes.

**Comment by hadley** [comment]
Could you please merge/rebase? I think I'm ready to start seriously looking at this.

We also need to move tests and move anything `data_frame()` related in the tibble news.

**Comment by krlmlr** [comment]
Done. I think tests may stay in both places for now; tibble test coverage is good already, but keeping the tests in dplyr makes sure that the original functionality remains.

**Comment by krlmlr** [comment]
Rebased, deprecated `add_rownames()`, re-added `dim_desc()`. Had to adapt test due to changed semantics of `type_sum()` for data frames. Tests pass locally.

Will remove the "Remotes" entry once tibble is on CRAN. Otherwise ready for review.

File: R/tibble-reexport.r [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# tbl-df ------------------------------------------------------------------
+
+#' @importFrom tibble tbl_df
+#' @export
+tibble::tbl_df
```
Comment (hadley) [inline R/tibble-reexport.r]: It's a bit late now, but I realised that `tbl_df` probably should've stayed in dplyr. It's not a big deal though, as `tbl_df()` is basically deprecated in favour of `as_data_frame()`

**Comment by hadley** [comment]
I think this looks good. Can you please add a bullet to NEWS?

What do you think about removing the tibble related tests now that tibble has 100% test coverage?

File: R/tibble-reexport.r (hunk 2 above)
Comment (krlmlr) [inline R/tibble-reexport.r]: No big deal, I can make a point release.

**Comment by hadley** [comment]
Actually don't worry about NEWS - I'm starting to do the pre-release cleanup

**Pull request closed without merging**